bootstrap/src/contexts/WorkspaceList.js

```diff
@@ -0,0 +1,62 @@
+import React, { createContext, useState, useContext, useEffect } from "react";
+import last from "lodash/last";
+
+import { defaultSourceState } from "../utils";
+import { useBackend } from "../contexts/Auth";
+import { useAuth } from "../contexts/Auth";
+import { AuthStatus } from "../contexts/Auth/state";
+import { workspaceFilter } from "../constants";
+
+const WorkspaceListContext = createContext({
+    ...defaultSourceState,
+    firstWorkspace: null,
+});
+
+const filterWorkspaces = (workspaces, filter) => {
+    return !filter ? workspaces : workspaces.filter(workspace => workspace.meta.title.match(filter));
+};
+
+const getFirstWorkspace = workspaces => {
+    return workspaces.length && last(workspaces[0].id.split("/"));
+};
+
+export const WorkspaceListProvider = ({ children }) => {
+    const { authStatus } = useAuth();
+    const backend = useBackend();
+    const [workspaceListState, setWorkspaceListState] = useState({ ...defaultSourceState });
+    const [firstWorkspace, setFirstWorkspace] = useState(null);
+
+    useEffect(() => {
+        const getWorkspaces = async () => {
+            setWorkspaceListState({ isLoading: true });
+            try {
+                const currentWorkspaces = (
```

Comment:
Careful, this does not necessarily return all the workspaces, this API is paged. Is this ok here? I think we should get all the pages, right?

Reply:
Unfortunately you would have to run through the API page by page until the items it returns are empty.

Reply:
The limit cannot be increased